File: php/src/Model/BulkTaskError.php
<?php

declare(strict_types=1);

namespace Fivexer\SDK\Model;

/**
 * Why one entry of a bulk create failed.
 *
 * Response model: built from a decoded JSON array and read-only thereafter.
 */
final class BulkTaskError
{
    public function __construct(
        public readonly string $code,
        public readonly string $message,
    ) {
    }

    /**
     * @param array<string, mixed> $data
     * @internal
     */
    public static function fromArray(array $data): self
    {
        return new self(
            code: (string) ($data['code'] ?? ''),
            message: (string) ($data['message'] ?? ''),
        );
    }
}

File: java/src/main/java/io/fivexer/sdk/model/BulkTaskError.java
package io.fivexer.sdk.model;

/**
 * Why one entry of a bulk create failed.
 *
 * <p>Populated by Gson via field reflection and exposed read-only; callers receive instances
 * from the client and never construct them.
 */
public class BulkTaskError {
    private String code;
    private String message;

    public String getCode() { return code; }
    public String getMessage() { return message; }
}
